crates/dojo-core/src/world.cairo

```diff
@@ -219,15 +336,61 @@ mod World {
         }
     }
 
-    // Returns entity IDs and entities that contain the component state.
+    /// Returns entity IDs and entities that contain the component state.
+    ///
+    /// # Arguments
+    ///
+    /// * `component` - The name of the component to be retrieved
+    /// * `partition` - The partition to be retrieved
+    ///
+    /// # Returns
+    ///
+    /// * `Span<u250>` - The entity IDs
+    /// * `Span<Span<felt252>>` - The entities
     #[view]
     fn entities(component: ShortString, partition: u250) -> (Span<u250>, Span<Span<felt252>>) {
         let class_hash = component_registry::read(component);
         Database::all(class_hash, component.into(), partition)
     }
 
+    /// Set the executor contract address
+    ///
+    /// # Arguments
+    ///
+    /// * `contract_address` - The contract address of the executor
     #[external]
     fn set_executor(contract_address: ContractAddress) {
+        // Only Admin can set executor
+        assert(is_account_admin(), 'only admin can set executor');
         executor::write(contract_address);
     }
+
+    /// Set the execution role for a system
+    ///
+    /// # Arguments
+    ///
+    /// * `system` - The name of the system
+    /// * `role_id` - The role id of the system
+    #[external]
+    fn set_execution_role(system: ShortString, role_id: u250) {
+        // Only Admin can set Admin role 
+        if role_id == 'Admin'.into() {
+            assert(is_account_admin(), 'only admin can set Admin role');
+        }
+        execution_role::write(system, role_id);
```

Comment:
hmm do we need the `system` here?

i think every system execute call could be proceeded by a `assume_role` call and i think we want to validate that the caller has permission to assume the role here?

that way, if there are multiple executions in a multicall, the caller can, for example, assume a role once that can be used across all the execute calls

Reply:
You're right good catch, I'm removing the system here and we just check if the caller has this permission to assume the role 👍 also this way, the role can be used across multiple system calls